src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -142,131 +147,239 @@ class CrossDeviceLinkUI extends Component {
+          </div>
+          <div className={style.numberInputSection}>
+            <div
+              className={classNames(style.inputContainer, {
+                [style.fieldError]: invalidNumber
+              })}
+            >
+              <PhoneNumberInputLazy
+                {...this.props}
+                clearErrors={this.clearErrors}
+              />
+            </div>
+            <Button
+              ariaLive="polite"
+              ariaBusy={sending}
+              className={classNames(style.btn, { [style.sending]: sending })}
+              variants={['primary']}
+              onClick={this.handleSendSmsLinkClick}
+              disabled={sending || invalidNumber}
+            >
+              {translate(buttonCopyKey)}
+            </Button>
+          </div>
+        </div>
+        <div role="alert" hidden={!invalidNumber}>
+          {invalidNumber && (
+            <div className={style.numberError}>
+              {translate('errors.invalid_number.message')}
+            </div>
+          )}
+        </div>
+      </div>
+    )
+  }
+
+  renderCopyLinkSection = () => {
+    const { translate } = this.props
+    const mobileUrl = this.getMobileUrl()
+    const { copySuccess } = this.state
+    const linkCopyKey = copySuccess ?
+      'cross_device.link.copy_link.success' :
+      'cross_device.link.copy_link.action'
+    return (
+      <div className={style.copyLinkSection}>
+        <div className={style.label}>
+          {translate('cross_device.link.copy_link_label')}
+        </div>
+        <div
+          className={classNames(
+            style.linkContainer,
+            copySuccess && style.copySuccess
+          )}
+        >
+          <span
+            className={style.linkText}
+            ref={element => (this.linkText = element)}
+          >
+            {mobileUrl}
+          </span>
+          {document.queryCommandSupported('copy') && (
+            <div className={style.actionContainer} aria-atomic="true" aria-live="polite">
+              <button
+                type="button"
+                onClick={() => copyToClipboard(mobileUrl, this.onCopySuccess)}
+                className={style.copyToClipboard}
+              >
+                {translate(linkCopyKey)}
+              </button>
+            </div>
+          )}
+        </div>
+        <hr className={style.divider} />
+      </div>
+    )
+  }
+
+  renderQrCodeSection = () => {
+    return (
+      <div className={style.qrCodeSection}>
+        <div className={style.qrCodeContainer}>
+          <QRCodeGenerator url={this.getMobileUrl()} size={144} />
+        </div>
+        <QRCodeHowTo />
+      </div>
+    )
+  }
+
+  handleViewOptionSelect = (newViewId) => {
+    sendEvent(`${newViewId.replace('_',' ')} selected`)
+    this.setState({ currentViewId: newViewId })
+  }
+
   componentWillUnmount() {
     this.clearSendLinkClickTimeout()
   }
 
+  // componentDidUpdate(prevProps, prevState) {
```

Comment:
What's happening with this comment?

Reply:
Oops. Forgot to remove this when I was done.

This is no longer needed since PageTitle (rightfully, given agreed screenreader behaviour) is handling the focus when its props update.